analyses/ssgsea-hallmark/02-analyze-plot-ssgsea-res.R

```diff
@@ -0,0 +1,148 @@
+
+#Choose Clinical Samples corresponding to samples in Gene Set Matrix
+clinData <- clinData[clinData[,"experimental_strategy"]=="RNA-Seq",]
+rownames(clinData) <- clinData[,"Kids_First_Biospecimen_ID"]
+clinData <- clinData[colnames(geneSetExpMat),]
+
+getVarNames <- function(x)
+{
+	tmp <- strsplit(x, split="-")[[1]]
+	tmp1 <- tmp[1]
+	tmp2 <- tmp[2]
+	return(c(tmp1, tmp2))
+}
+
+#Analyse Pathways statistically correlated with clinical variables
+#Returns ANOVA P-value and Tukey HSD Results
+#Requires Pathway (pathwayName) and Clinical Variable (clinVarName)
+compareClinVarToPathway <- function(pathwayName=NULL, clinVarName=NULL)
+{
+	tmpDat <- data.frame(clinData[,clinVarName], geneSetExpMat[pathwayName,])
+	colnames(tmpDat) <- c("ClinVar", "PathwayVar")
+	aovOut <- aov(PathwayVar ~ ClinVar, data=tmpDat)
+	posthoc <- TukeyHSD(x=aovOut, conf.level=0.95)
+	posthoc <- data.frame(posthoc[[1]])
+	posthoc[,"F_Value"] <- summary(aovOut)[[1]][[4]][1]
+	posthoc[,"AOV_P"] <- summary(aovOut)[[1]][[5]][1]
+	posthoc[,"Pathway"] <- pathwayName
+	posthoc[,"ClinVar"] <- clinVarName
+	posthoc <- cbind(t(sapply(rownames(posthoc), FUN=getVarNames)), posthoc)
+	colnames(posthoc)[1:2] <- c("VarX", "VarY")
+	return(posthoc)
+}
+
+##############################
+#Analysis by Disease
+##############################
+diseaseType <- lapply(rownames(geneSetExpMat), FUN=compareClinVarToPathway, clinVarName="short_histology")
+diseaseType <- do.call("rbind", diseaseType)
+
+#Filter to significant entries
+diseaseTypeFilt <- diseaseType %>% dplyr::filter( AOV_P <0.01, p.adj < 0.05)
+
+#Now make all comparisons the same direction & write out results
+diseaseTypeFilt[,"DiseaseX"] <- ifelse(diseaseTypeFilt[,"diff"]>0, as.character(diseaseTypeFilt[,"VarX"]), as.character(diseaseTypeFilt[,"VarY"]))
+diseaseTypeFilt[,"DiseaseY"] <- ifelse(diseaseTypeFilt[,"diff"]>0, as.character(diseaseTypeFilt[,"VarY"]), as.character(diseaseTypeFilt[,"VarX"]))
+diseaseTypeFilt <- diseaseTypeFilt[-1:-2]
+diseaseTypeFilt <- diseaseTypeFilt[, c("DiseaseX", "DiseaseY", colnames(diseaseTypeFilt)[1:8])]
+write.table(diseaseTypeFilt, "results/DiseaseCorrelationPathway.txt", sep="\t", row.names=F)
+
+#Now get diseases highly associated with a certain pathway
+diseaseTableTmp <- data.frame(table(diseaseTypeFilt[,c("DiseaseX", "Pathway")]))
+diseaseTableTmp <- diseaseTableTmp[diseaseTableTmp[,"Freq"]>(nrow(geneSetExpMat)*.25),]
+diseaseTableTmp <- diseaseTableTmp[order(-diseaseTableTmp[,"Freq"]),]
+
+#Now set up to create heatmaps
+keepPathways <- unique(diseaseTableTmp[,"Pathway"])
+keepDisease <- unique(diseaseTableTmp[,"DiseaseX"])
+tmpClinData <- clinData[c("short_histology", "broad_histology")]
+colnames(tmpClinData)[1] <- "Histology" 
+rownames(geneSetExpMat) <- gsub("HALLMARK", "", rownames(geneSetExpMat))
+rownames(geneSetExpMat) <- gsub("_", " ", rownames(geneSetExpMat))
+rownames(geneSetExpMat) <- trimws(rownames(geneSetExpMat))
+
+#Heatmap across all samples, only significant pathways
```

Comment:
Separate logic to write out results in CI vs otherwise is brittle. There's code that's not being tested. If you set things from the command line (see above), you could just make the significance thresholds very permissive and then you'd be testing the full script.

Reply:
I can do this, I'm just not sure how to make the threshold different for CircleCI and for the Repo, i.e. if someone wanted to just run the whole thing.

Reply:
It looks like you're already using a shell script to run this. You can set defaults in that shell script that you think should be used in general. Then, when you call that shell script via CI you should pass more permissive parameters so that the full code path can be tested.

You can set defaults like so:
https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/download-data.sh#L6

Then you can override in CI with this approach:
https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/.circleci/config.yml#L13

Reply:
Let's get some feedback from @jaclyn-taroni - as the architect of `run-in-ci.sh` there may be some differences for commands run using that.

Reply:
Sure, @jaclyn-taroni let me know the best approach whenever you have some time.

Reply:
Hi @PichaiRaman - I had a bit of time yesterday to test a few things out, and I needed to add just a little bit of functionality to support what I've described above. I've filed #147 to implement that functionality and add a description of how to do this to the README. I'd love to get your feedback on #147 changes - especially the documentation ones. Once that lands we can easily address this concern by using a permissive threshold here.

Reply:
Thanks @cgreene. The documentation is very clear, I'm just going to test it out.

Reply:
Exciting! You'll have to replace run_in_ci.sh with the file from my branch to have the described approach work. This should be a direct link to the download:
https://github.com/AlexsLemonade/OpenPBTA-analysis/raw/23d231d06b889f951f10e9c096a1159f610ea9e8/scripts/run_in_ci.sh